src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -146,17 +150,20 @@ class LibraryService (protected val argUserInfo: UserInfo,
   def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
     rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
       val pub = isPublished(workspaceResponse)
+      val invalidMetadata = workspaceResponse.workspace.attributes.get(
+        AttributeName(AttributeName.libraryNamespace, "invalidDataset")).orElse(Some(AttributeBoolean(false)))
```

Comment:
Nit ... a line break after the open paren is confusing here. Maybe this is more legible?
```
      val invalidMetadata = workspaceResponse.workspace.attributes.
        get(AttributeName(AttributeName.libraryNamespace, "invalidDataset")).
        orElse(Some(AttributeBoolean(false)))
```